orchestrator-salt/src/main/resources/salt/salt/nginx/conf/clouderamanager-ssl-user-facing.conf

```diff
@@ -28,6 +28,14 @@ server {
         rewrite ^([/](?!clouderamanager/).*$) /clouderamanager$1;
     }
 
+    # If hadoop-jwt cookie not set and favicon is requested return 403.
+    if ($cookie_hadoop-jwt !~ .+) {
+        rewrite ^/favicon.ico$ /favicon.403;
+    }
+    location ~ ^/favicon.403$ {
+        return 403;
+    }
```

Comment:
Why not just add the location to the first if statement?
if ($cookie_hadoop-jwt .. && location favicon.ico) return 403?

Reply:
Oh I wish ! quirks of nginx. 
Location is not allowed in `if `and `if` is frowned upon in Location, see [If is Evil ](https://www.nginx.com/resources/wiki/start/topics/depth/ifisevil/)